server/src/main/java/org/apache/druid/server/coordinator/RandomBalancerStrategy.java

```diff
@@ -26,24 +26,20 @@
 import java.util.Iterator;
 import java.util.List;
 import java.util.NavigableSet;
-import java.util.concurrent.ThreadLocalRandom;
 import java.util.stream.Collectors;
 
 public class RandomBalancerStrategy implements BalancerStrategy
 {
   @Override
-  public ServerHolder findNewSegmentHomeReplicator(DataSegment proposalSegment, List<ServerHolder> serverHolders)
+  public Iterator<ServerHolder> findNewSegmentHomeReplicator(DataSegment proposalSegment, List<ServerHolder> serverHolders)
```

Comment:
Sometimes returning an `Iterator<>` is right, this signature feels like we might as well just return the `List<>` instead?

Reply:
Sounds good.

Reply:
On second thought, I wonder if `Iterator` is actually better. A caller should always iterate over the eligible servers in order and we get the benefit of the simple `hasNext` and `next` methods, rather than having to use `isEmpty` and `get`.

The one benefit I see with having a list is if we wanted to get the count of servers that were found to be eligible upfront (in case we wanted to report that).
What do you advise?

Reply:
I don't feel overly strong about it.  Was more of a stream-of-conciousness/feeling comment.  So now that you've thought about it directly and like Iterator better, let's keep that.